### PR TITLE
Allow passing options to toggleComment

### PR DESCRIPTION
**Pull request by vincentwoo**
The `toggleComment` helper is very useful, and the sublime keymap uses it by default. However, the actual sublime behavior for that keystroke is to comment with indentation.

This change makes `toggleComment` a first class extension method, and leaves the command (of the same name) to just call the extension. The new extension also takes an options object to allow more customization.

### CODE REVIEW

File: addon/comment/comment.js [hunk 1]
```diff
@@ -21,22 +21,22 @@
...
+
+  CodeMirror.defineExtension("toggleComment", function(options) {
+    if (!options) options = noOptions;
+    var minLine = Infinity, ranges = this.listSelections();
     for (var i = ranges.length - 1; i >= 0; i--) {
       var from = ranges[i].from(), to = ranges[i].to();
       if (from.line >= minLine) continue;
       if (to.line >= minLine) to = Pos(minLine, 0);
       minLine = from.line;
-      if (mode == null) {
```
Comment (marijnh) [inline addon/comment/comment.js]: Why did you remove the mode-preserving logic here?

**Comment by vincentwoo** [comment]
@marijnh I deleted the mode stuff because I am a big idiot who codes tired. I restored it - I thought the mode code could never do anything because `mode` was always `null`... hah.

**Comment by marijnh** [comment]
Merged as 8a012c0ae93654, with the change of binding `cm` to `this` in the extension method, so that it actually works.

**Comment by vincentwoo** [comment]
oh god, good catch, sorry
